codenames-domain/src/main/java/com/codenames/services/RoomService.java

```diff
@@ -0,0 +1,149 @@
+package com.codenames.services;
+
+import com.codenames.dto.RoomDto;
+import com.codenames.enums.GameStatus;
+import com.codenames.enums.GameTurn;
+import com.codenames.enums.PlayerRole;
+import com.codenames.mapper.RoomMapper;
+import com.codenames.models.game.Player;
+import com.codenames.models.room.Room;
+import com.codenames.models.room.Team;
+import lombok.RequiredArgsConstructor;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@RequiredArgsConstructor
+public class RoomService {
+
+    private final TeamService teamService;
+
+    private final WordsService wordsService;
+
+    private final RoomMapper roomMapper = Mappers.getMapper(RoomMapper.class);
```

Comment:
a simple constructor-based dependency injection should do the trick, maybe with some mapper modifications (see https://www.baeldung.com/mapstruct#1-modify-the-mapper)

Reply:
Yes, you are right, we could have made this initialization a little more convenient